### DISCUSSION TITLE
Disable extensions on migrations or schema apply

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#15015] [Archive]
Hi!

We run our Directus in a containerized, selfscaling, environment on [Google Cloud Platform](https://docs.directus.io/self-hosted/installation/gcp/).

Since this environment scales back its instances on no usage, it will spin up a new instance if Directus is touched after a while.

Now in our Docker container we do three things:
- Migrate the database to the latest (will usually say "Database up to date)
- Apply our schema (will usually say "No changes to apply.)
- We start directus

Now for each step, all extensions will be loaded. And since a few of these extensions do external checks on "server.start" it influences our cold start time.

Now the question is, to reduce this cold start time: can we disable the extension loading in the first two cases? We don't need them there!

Any help is greatly appreciated!

Kees.

**Comment by keesvanbemmel** [comment]
While my question about disabling extensions in certain usecases still stands, I solved my specific case by only running migrations/schema application when a certain ENV var is set, resetting it after this is done.

Cheers!

**Reply by azrikahar** [reply]
There is a discussion on a potential flag to skip custom migrations as mentioned here: https://github.com/directus/directus/discussions/11798#discussioncomment-2267274, so perhaps there can also be an additional flag for disabling extensions when that idea gets explored further.
